cicecore/cicedyn/infrastructure/ice_domain.F90

```diff
@@ -216,21 +216,36 @@ subroutine init_domain_blocks
    call broadcast_scalar(maskhalo_bound,    master_task)
    call broadcast_scalar(add_mpi_barriers,  master_task)
    call broadcast_scalar(debug_blocks,      master_task)
-   if (my_task == master_task) then
-     if (max_blocks < 1) then
-       max_blocks=( ((nx_global-1)/block_size_x + 1) *         &
-                    ((ny_global-1)/block_size_y + 1) - 1) / nprocs + 1
-       max_blocks=max(1,max_blocks)
-       write(nu_diag,'(/,a52,i6,/)') &
-         '(ice_domain): max_block < 1: max_block estimated to ',max_blocks
-     endif
-   endif
    call broadcast_scalar(max_blocks,        master_task)
    call broadcast_scalar(block_size_x,      master_task)
    call broadcast_scalar(block_size_y,      master_task)
    call broadcast_scalar(nx_global,         master_task)
    call broadcast_scalar(ny_global,         master_task)
 
+   ! Set nprocs if not set in namelist
+#ifdef CESMCOUPLED
+   nprocs = get_num_procs()
+#else
+   if (nprocs == -1) then
+      nprocs = get_num_procs()
+   else if (nprocs /= get_num_procs()) then
+      write(nu_diag,*) subname,' ERROR: nprocs, get_num_procs = ',nprocs,get_num_procs()
+      call abort_ice(subname//' ERROR: Input nprocs not same as system request', file=__FILE__, line=__LINE__)
+   endif
+#endif
+
+   ! Determine max_blocks if not set
+   if (max_blocks < 1) then
+      max_blocks=(((nx_global-1)/block_size_x + 1) *         &
+                  ((ny_global-1)/block_size_y + 1) - 1) / nprocs + 1
+      max_blocks=max(1,max_blocks)
+      if (my_task == master_task) then
+         write(nu_diag,'(/,a52,i6,/)') &
+            '(ice_domain): max_block < 1: max_block estimated to ',max_blocks
+      endif
+   endif
+   
+   
```

Comment:
can we avoid whitespace-only lines here ? Also I would think a single blank line is sufficient.

Reply:
I made it one line instead of two :)